circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/DiffGeneratedPartitionPredicate.java

```diff
@@ -55,19 +57,23 @@ public DiffGeneratedPartitionPredicate(
     this.replica = replica;
     this.tableReplication = tableReplication;
     this.checksumFunction = checksumFunction;
+    if (tableReplication.getSourceTable().getPartitionLimit() != null) {
+      partitionLimit = tableReplication.getSourceTable().getPartitionLimit();
+    }
   }
 
   private String generate() {
     try (CloseableMetaStoreClient sourceMetastore = source.getMetaStoreClientSupplier().get()) {
       try (CloseableMetaStoreClient replicaMetastore = replica.getMetaStoreClientSupplier().get()) {
         Table sourceTable = source.getTableAndStatistics(tableReplication).getTable();
         PartitionIterator partitionIterator = new PartitionIterator(sourceMetastore, sourceTable,
-            tableReplication.getPartitionIteratorBatchSize());
+            tableReplication.getPartitionIteratorBatchSize(), Ordering.REVERSE);
```

Comment:
Isn't this also a change in behaviour that we should mention in the CHANGELOG? HiveDiff will now process the partitions with newest first while in the past it was oldest first?

Reply:
yeah I'll add a note